### PR TITLE
BREAKING(unstable): remove CreateHttpClientOptions.caFile

### LINKED ISSUE #8837: Remove caFile from Deno.createHttpClient options
`Deno.createHttpClient` now can be configured to use a CA using the `caData` field. They `caFile` field is this not needed anymore.

### PR DESCRIPTION
**Pull request by mrmodise**
Closes #8837

### CODE REVIEW

**Comment by mrmodise** [comment]
@lucacasonato @bartlomieju would this change affect passing the cert as a file on the terminal. I notice the tests that are failing all have a similar pattern: ```--cert tls/RootCA.pem```

For example,

```rust
itest!(cafile_info {
  args:
    "info --quiet --cert tls/RootCA.pem https://localhost:5545/cli/tests/cafile_info.ts",
  output: "cafile_info.ts.out",
  http_server: true,
});
```

Any feedback is welcome. Thanks

**Comment by bartlomieju** [comment]
@mrmodise no, `--cert` flag should still work as before

**Comment by mrmodise** [comment]
Thanks @bartlomieju I thought they were linked cause when i checked the `program_state.rs` I see that it accepts the create_http_client() as shown below meaning I would have to write another function to read the cert into a string

```rust
let file_fetcher = FileFetcher::new(
      http_cache,
      cache_usage,
      !flags.no_remote,
      ca_file.as_deref(),
    )?;
```

since removed dealing with a file as shown below:

```rust
if let Some(ca_file) = ca_file {
    let mut buf = Vec::new();
    File::open(ca_file)?.read_to_end(&mut buf)?;
    let cert = reqwest::Certificate::from_pem(&buf)?;
    builder = builder.add_root_certificate(cert);
  }
```

**Comment by bartlomieju** [comment]
@mrmodise then I think it's safe to move that bit of code into `FileFetcher` to handle that situation

**Comment by mrmodise** [comment]
@bartlomieju I think its ready for review. I also wanted to say thanks once again for always getting back to my questions. Its really encouraging and inspiring for me as  a new joiner, and, am sure I speak for all new joiners who come to this repository to  add value to Deno in their own ways

File: cli/program_state.rs [hunk 1]
```diff
@@ -58,7 +59,13 @@ impl ProgramState {
     let dir = deno_dir::DenoDir::new(custom_root)?;
     let deps_cache_location = dir.root.join("deps");
     let http_cache = http_cache::HttpCache::new(&deps_cache_location);
-    let ca_file = flags.ca_file.clone().or_else(|| env::var("DENO_CERT").ok());
+    let ca_file_path =
+      flags.ca_file.clone().or_else(|| env::var("DENO_CERT").ok());
+
+    let ca_data: Option<String> = match ca_file_path.as_ref() {
+      None => None,
+      Some(ca_file_path) => Some(read_to_string(ca_file_path).unwrap()),
```
Comment (bartlomieju) [inline cli/program_state.rs:67]: This bit will panic if file doesn't exist, use `?` instead of `unwrap()`

**Review by bartlomieju — CHANGES_REQUESTED** [review]
@mrmodise thanks, looks good, just one thing to fix

File: cli/program_state.rs (hunk 1 above)
Comment (mrmodise) [reply cli/program_state.rs]: Cool thought of it but I was not that sure. I never worked with Rust before this PR but am definitely hooked

**Review by bartlomieju — APPROVED** [review]
LGTM

**Pull request merged**